packages/memory/src/memory-fs.ts

```diff
@@ -301,15 +352,46 @@ export function createBaseMemoryFsSync(): IBaseMemFileSystemSync {
     return { isFile, isDirectory, isSymbolicLink, birthtime, mtime };
   }
 
+  function lstatSync(nodePath: string): IFileSystemStats {
+    const resolvedPath = resolvePath(nodePath);
+    const node = getNode(resolvedPath, false);
+    if (!node) {
+      throw createFsError(resolvedPath, FsErrorCodes.NO_FILE_OR_DIRECTORY, 'ENOENT');
+    }
+    const { birthtime, mtime } = node;
+    const isFile = node.type === 'file' ? returnsTrue : returnsFalse;
+    const isDirectory = node.type === 'dir' ? returnsTrue : returnsFalse;
+    const isSymbolicLink = node.type === 'symlink' ? returnsTrue : returnsFalse;
+
+    return { isFile, isDirectory, isSymbolicLink, birthtime, mtime };
+  }
+
   function realpathSync(nodePath: string): string {
     const resolvedPath = resolvePath(nodePath);
-    const node = getNode(resolvedPath);
+    const node = getNode(resolvedPath, false);
     if (!node) {
       throw createFsError(resolvedPath, FsErrorCodes.NO_FILE_OR_DIRECTORY, 'ENOENT');
     }
+
+    if (node.type === 'symlink') {
+      const { path } = getRealNode(node);
+      return path;
+    }
     return resolvedPath;
   }
 
+  function getRealNode(node: IFsMemSymlinkNode): { node: IFsMemNodeType | undefined; path: string } {
+    const resolvedLink = getNode(node.path, false);
```

Comment:
@AviVahl I am not following the symlinks because I want to follow the path for each node I visited, to return it